text/0000-crates.io-user-namespaces.md

```diff
@@ -0,0 +1,169 @@
+- Feature Name: crates_io_user_namespaces
+- Start Date: 2020-08-29
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes extending the crates.io package naming format to allow
+per-user namespaces. A crates.io user Jane Doe signed in with the GitHub
+account `"jdoe"` would be able to publish crates with names prefixed by
+`"~jdoe/"` Tooling that operates on crates.io package names, such as docs.rs,
+will recognize this syntax and drop the prefix to compute the crate name.
+
+# Motivation
+[motivation]: #motivation
+
+crates.io currently places all registered packages in a single flat namespace.
+This simplifies some aspects of package discovery, but adds friction for users
+who prefer to name packages after their purpose rather than use an opaque
+codename. Additionally, due to the large number of registered packages, it is
+often difficult to find a name that is short enough to be ergonomic when
+used as a symbol in Rust source code.
+
+Introducing per-user namespaces will allow every crates.io user to upload packages without worrying about crate name conflicts.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+TODO: Is a guide-level explanation needed at all? There's already documentation
+on how to define dependencies in Cargo where the package name is different from
+the crate name, and this isn't adding any new capabilities.
```

Comment:
I would like to see what a person needs to do to add a user-namespaced dependency to their project, and how they would use it in their source code. I can roughly figure it out by reading the rest of the RFC, but I'm not sure it's exactly what you had in mind :)

Reply:
Cargo supports specifying the package name, so something like this (copying from https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html):
```toml
[dependencies]
foo = { version = "0.1", package = "~someuser/foo" }
```

Reply:
Could you add that to the guide level explaination? Something like "the user visits the crate's github page, copies this snippet from the readme [snippet], and uses the crate in the code like this [snippet]". I feel like showing the whole process a user needs to do to use a crate could help people form their opinion on the RFC.